### ISSUE TITLE
Join forces?

### ISSUE THREAD

**Issue by thedodd** [datafuse-extras/async-raft#4]
Hello datafuse team! I just wanted to open a channel of communication and see if you all would be interested in getting this fork merged upstream. I know it is 111 commits difference at this point, but taking the changes and adding some folks as co-maintainers would be awesome for the overall health of the async-raft project.

Just let me know, cheers 🍻

**Comment by drmingdrmer** [comment]
Feel free to merge back every of the commits back to upstream!

**Comment by semirix** [comment]
@drmingdrmer Is it important for datafuse to maintain their own separate fork or would you be open to being an upstream maintainer?

**Comment by drmingdrmer** [comment]
> @drmingdrmer Is it important for datafuse to maintain their own separate fork or would you be open to being an upstream maintainer?

It depends. 
For now we need to move quickly improving features or fixing bugs.

And to be honest, there are some personal style in the latest commits. 🤔 
E.g. I merged most `term, index` tuple into a `LogId`.

And I've changed some of the API that make it easy to use with a real storage(`sled` based as @thedodd suggested).

And there will be some optimization that is not included in raft spec.
E.g. the membership representation. And the messy leadership management.
